### ISSUE TITLE
Asset caching doesn't restore

### ISSUE THREAD

**Issue by cmannett85** [microsoft/vcpkg#20725]
**Describe the bug**
I'm using filesystem-based asset caching in my CI system in conjunction with NuGet-based binary caching (which works fine).  It looks like asset caching is working in so far that the assets are being correctly stored in the specified location, but if there's a binary cache miss the package source is still being downloaded from the public mirrors rather than the asset cache.

In my case I'm zipping up the asset cache dir and pushing to Artifactory at the end of a successful build, and downloading and extracting at the start of the next.  As an example I deleted an openssl package from my NuGet repo, the package was dutifully rebuilt but using the public mirror.
```
// export X_VCPKG_ASSET_SOURCES="clear;x-azurl,file:///home/jenkins/agent/workspace/streamline-aws/branch-develop/pre-commit/change-364167/build/vcpkg_asset_cache,,readwrite"
Asset cache hit, unzipping to /home/jenkins/agent/workspace/streamline-aws/branch-develop/pre-commit/change-364167/build/vcpkg_asset_cache
Archive:  asset_cache.zip
  inflating: /home/jenkins/agent/workspace/streamline-aws/branch-develop/pre-commit/change-364167/build/vcpkg_asset_cache/637f6d436614a805bb7d8e5dac0fc154896f4e77a51e1bf2d4509779bc412eaff1756dedcc9937490ca21d5bc15c8df6d911cfe9eebce51f80b5ec42a85d4155  
...
Attempting to fetch 60 packages from nuget.
Restored 59 packages from NuGet in 28.98 s. Use --debug for more information.
Starting package 1/60: openssl:gatord-armv7-linux-musl
Building package openssl[core]:gatord-armv7-linux-musl...
-- [OVERLAY] Loading triplet configuration from: /home/jenkins/agent/workspace/streamline-aws/branch-develop/pre-commit/change-364167/build/native/gator/daemon/cmake/triplets/gatord-armv7-linux-musl.cmake
-- Downloading https://www.openssl.org/source/openssl-1.1.1l.tar.gz;https://www.openssl.org/source/old/1.1.1/openssl-1.1.1l.tar.gz -> openssl-1.1.1l.tar.gz...
```
I've also noticed that specifying  `x-block-origin;` before `azurl` source makes no difference - the public mirrors are still used.

**Environment**
- OS: Linux in a Docker container
- Compiler: n/a

**To Reproduce**
1. Set the `X_VCPKG_ASSET_SOURCES` env var to `"clear;x-azurl,file:///my-asset-cache,,readwrite"
2. Do a build to populate the asset cache
3. Delete some built NuGet packages from your binary cache to force a rebuild of them
4. Rebuild and observe that the rebuilt packages download from public mirrors, not the asset cache

**Expected behavior**
vcpkg will use the filesystem-based asset cache to build a package from rather than the public mirrors.

**Failure logs** 
Sorry can't, commercially sensitive.

**Comment by cmannett85** [comment]
> Tl;dr: The message is lying to you.
Ah, I suspected as much but I couldn't just turn the network off on my corporate CI system!  I didn't know about the proxy options.
